src/main/java/com/nutomic/syncthingandroid/syncthing/SyncthingRunnable.java

```diff
@@ -22,39 +22,77 @@
 
     private static final String TAG_NATIVE = "SyncthingNativeCode";
 
+    public static final String UNIT_TEST_PATH = "was running";
+
     private final Context mContext;
 
-    private String mCommand;
+    private boolean mGenerate;
+
+    private Command mCommand;
+
+    private String mManualCommand;
 
+    public enum Command {
+        generate, // Generate keys, a config file and immediately exit.
+        main,     // Run the main Syncthing application.
+        reset,    // Clear the Syncthing index.
+        manual,   // Manual command
```

Comment:
Deleting app data resets your device id as well. reset only clears the indexes and moves all folders to a backup location